plan/views.py

```diff
@@ -54,35 +56,47 @@ def get_sections(data):
     return sections
 
 
+def check_semester(data, sections):
```

Comment:
might be nice to also return the semester it sets data's semester to, in case we use this function for something different going forward

Reply:
The problem is 
```
semester_res = self.check_semester(data, sections)
        if semester_res is not None:
            return semester_res
```
If the function were not partial, then this code wouldn't work...